homeassistant/components/rest/sensor.py

```diff
@@ -193,6 +195,11 @@ def update(self):
 
             self.data = response.text
         except requests.exceptions.RequestException as ex:
-            _LOGGER.error("Error fetching data: %s from %s failed with %s",
-                          self._request, self._request.url, ex)
-            self.data = None
+            try:
+                response = requests.get(self._resource, headers=self._headers,
```

Comment:
this makes no sense. You're retrying to fetch and do it on every type of error.

Reply:
Automatic retry is scary. Maybe we should just remove the `self.data = None` assignment on failures?

And maybe the `sensor.scrape` timeout should be configurable, like it is with `sensor.rest`.

Reply:
This call is all wrong. Verify SSL is not passed in. It's using the global session from requests. 

Retries should not just copy paste the code for the number of retries we want